src/task/commits/commits.ts

```diff
@@ -0,0 +1,50 @@
+import { task } from '../../task.js'
+import { paginate } from '../../utils.js'
+import { CommitsQuery } from './commits.graphql.js'
+
+export default task({
+  name: 'commits' as const,
+  run: async (
+    { octokit, data, next },
+    { owner, name }: { owner: string; name: string },
+  ) => {
+    console.log(`Loading commits for ${owner}/${name}`)
+    const commits = paginate(octokit, CommitsQuery, {
+      owner: owner,
+      name: name,
+      author: data.user.id,
+    })
+
+    for await (const resp of commits) {
+      const { totalCount, nodes } =
+        resp.repository?.defaultBranchRef?.target?.history!
+
+      if (!nodes) {
+        throw new Error('Failed to load commits')
+      }
+
+      if (totalCount >= 10_000) {
```

Comment:
Should this limit be configurable?

Reply:
I don't think so. Only very strange repos, like homebrew, have more than 10k commit.s